### PR TITLE
Update py-meson-python (0.11.0, 0.12.0) and meson (0.64.1, 1.0.0)

### PR DESCRIPTION
**Pull request by rgommers**
Also add myself as maintainer for `py-meson-python`.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-meson-python/package.py
Comment (rgommers) [inline var/spack/repos/builtin/packages/py-meson-python/package.py]: This changed in `0.12.0`. I considered splitting this into two lines, but it doesn't really matter and I wasn't sure about the `when` syntax for versions of both python and meson-python, so I decided to do the simpler thing.

**Review by michaelkuhn — DISMISSED** [review]
LGTM (waiting for feedback from the py-meson-python maintainers before merging)

File: var/spack/repos/builtin/packages/meson/package.py
Comment (eli-schwartz) [inline var/spack/repos/builtin/packages/meson/package.py:22]: Some time today, 1.0.0 is intended to be released.

**Review by eli-schwartz — DISMISSED** [review]
This seems correct and it makes sense for @rgommers to be one of the maintainers of py-meson-python.

File: var/spack/repos/builtin/packages/py-meson-python/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-meson-python/package.py]:
```suggestion
    # https://github.com/mesonbuild/meson-python/pull/111
```

File: var/spack/repos/builtin/packages/py-meson-python/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-meson-python/package.py]:
```suggestion
    depends_on("py-tomli@1:", when="^python@:3.10", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-meson-python/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-meson-python/package.py]: FWIW, the correct syntax would be:
```python
depends_on("py-typing-extensions@3.7.4:", when="@0.12: ^python@:3.9", type=("build", "run"))
depends_on("py-typing-extensions@3.7.4:", when="@:0.11 ^python@:3.7", type=("build", "run"))
```
I don't care that much about making it perfect, as long as it works.

File: var/spack/repos/builtin/packages/py-meson-python/package.py
Comment (eli-schwartz) [reply var/spack/repos/builtin/packages/py-meson-python/package.py]: This is possible since py-meson-python 0.11.0, not before.

File: var/spack/repos/builtin/packages/meson/package.py
Comment (eli-schwartz) [reply var/spack/repos/builtin/packages/meson/package.py:22]: BTW, it's out now.

File: var/spack/repos/builtin/packages/py-meson-python/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-meson-python/package.py]:
```suggestion
    depends_on("py-tomli@1:", when="@0.11: ^python@:3.10", type=("build", "run"))
    depends_on("py-tomli@1:", when="@:0.10", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/meson/package.py
Comment (rgommers) [reply var/spack/repos/builtin/packages/meson/package.py:22]: Awesome! Let me add that as well.

Comment (rgommers) [reply var/spack/repos/builtin/packages/meson/package.py:22]: And, congrats on the 1.0.0 release @eli-schwartz - that's a milestone!

File: var/spack/repos/builtin/packages/py-meson-python/package.py
Comment (rgommers) [reply var/spack/repos/builtin/packages/py-meson-python/package.py]: Thanks. Now that I have to update a PR, let me add this. The added detail may be useful perhaps, who knows.

File: var/spack/repos/builtin/packages/meson/package.py
Comment (eli-schwartz) [reply var/spack/repos/builtin/packages/meson/package.py:22]: https://nibblestew.blogspot.com/2022/12/after-exactly-10-years-meson-100-is-out.html

Fun stuff. :)

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Is https://github.com/mesonbuild/meson-python/pull/167 in a stable release now? If so we should update `py-scipy` to simplify the install procedure (in a separate PR).

**Comment by rgommers** [comment]
> Is [mesonbuild/meson-python#167](https://github.com/mesonbuild/meson-python/pull/167) in a stable release now? If so we should update `py-scipy` to simplify the install procedure (in a separate PR).

Yes that's in `0.11.0`. Agreed, let's try to use that to clean things up.

**Pull request merged**
